code/modules/detective_work/detective_work.dm

```diff
@@ -3,12 +3,29 @@
 /atom/proc/add_fibers(mob/living/carbon/human/M)
 	if(M.gloves && isclothing(M.gloves))
 		var/obj/item/clothing/gloves/G = M.gloves
-		if(G.transfer_blood > 1) //bloodied gloves transfer blood to touched objects
-			if(add_blood(G.blood_DNA, G.blood_color)) //only reduces the bloodiness of our gloves if the item wasn't already bloody
-				G.transfer_blood--
-	else if(M.bloody_hands > 1)
-		if(add_blood(M.blood_DNA, M.hand_blood_color))
+		if(isitem(src) && G.blood_DNA)
```

Comment:
Is there any way you could rework this to avoid doing what amounts to an `istype(src)` check?

Reply:
replaced with variable easy_to_spill_blood on atom. No istype check now, just extra variable on all atoms in game, i hope it works better